backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/PPRoutingProduct.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * #%L
+ * metasfresh-material-planning
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.material.planning.pporder;
+
+import de.metas.product.ProductId;
+import lombok.Builder;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+
+@Value
+@Builder(toBuilder = true)
+public class PPRoutingProduct
```

Comment:
consider annotating with `@NonNull` (lombok) what's not null. In this way lombok will insert NPE checks.